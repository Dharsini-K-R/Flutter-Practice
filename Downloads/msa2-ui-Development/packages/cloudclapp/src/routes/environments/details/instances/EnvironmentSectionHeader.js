import React from "react";
import { Box, Grid, makeStyles, Typography } from "@material-ui/core";
import PropTypes from "prop-types";
import UserButton from "cloudclapp/src/components/user-button";
import StatusBadgeIcon from "cloudclapp/src/components/status-badge-icon";
import { ICONS } from "cloudclapp/src/components/status-badge-icon";

const useStyles = makeStyles(({ typography }) => ({
  envName: (props) => ({
    fontSize: 18,
    marginLeft: props.configTab ? 0 : 6,
    fontWeight: 600,
    color: typography.body1.color,
  }),
  environmentTitleWrapper: {
    borderBottom: "1px solid #B2BCCE",
  },
}));

const EnvironmentSectionHeader = ({
  title,
  owner,
  status,
  isLoading = false,
  configTab = true,
}) => {
  const classes = useStyles({ configTab });
  return (
    <Grid
      item
      xs={12}
      container
      justifyContent="space-between"
      alignItems={"center"}
      className={classes.environmentTitleWrapper}
    >
      <Grid item>
        <Box display="flex" flexDirection="row" alignItems={"center"}>
          {!configTab && (
            <Box>
              <StatusBadgeIcon
                size="large"
                icon={ICONS.environment}
                status={status}
              />
            </Box>
          )}
          <Box p={1}>
            <Typography
              id="INSTANCE_DETAILS_ENVIRONMENT_NAME"
              className={classes.envName}
            >
              {title}
            </Typography>
          </Box>
        </Box>
      </Grid>
      <Grid item>
        <Box display="flex" flexDirection="row" alignItems={"center"}>
          {/* <Typography
                id={`instance_details_created_at`}
                className={`${classes.text} ${classes.ratingText}`}
              >
                {t("Updated")}
                {": "}
                {formatDateOrString(
                  "2022-03-08T02:44:50.565Z",
                  "dd MMM yyyy HH:mm:ss",
                )}{" "}
              </Typography> */}
          <Box p={1}>
            <UserButton
              id="INSTANCE_DETAILS_OWNER_BUTTON"
              userId={owner}
              disabled={isLoading}
            />
          </Box>
        </Box>
      </Grid>
    </Grid>
  );
};
EnvironmentSectionHeader.propTypes = {
  title: PropTypes.string,
  owner: PropTypes.number,
  status: PropTypes.string,
  isLoading: PropTypes.bool,
  configTab: PropTypes.bool,
};

export default EnvironmentSectionHeader;
