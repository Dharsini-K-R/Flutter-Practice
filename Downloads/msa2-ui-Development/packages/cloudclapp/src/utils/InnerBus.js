import EventEmitter from "events";

// List of all events available for usage with inner bus
const EVENTS = {
  LOGOUT: "LOGOUT",
  LOGIN: "LOGIN",
};

class InnerBus extends EventEmitter {
  evt = EVENTS;
}

export default new InnerBus();
