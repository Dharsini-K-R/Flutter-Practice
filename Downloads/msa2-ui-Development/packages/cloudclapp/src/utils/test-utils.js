import React from "react";
import { render } from "@testing-library/react";
import { renderHook, act } from "@testing-library/react-hooks";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import { StylesProvider } from "@material-ui/core";
import { ThemeProvider } from "@material-ui/core";
import SnackbarProvider from "cloudclapp/src/SnackbarProvider";
import createTheme from "cloudclapp/src/styles/theme";
import thunk from "redux-thunk";
const middlewares = [thunk];

const msaTheme = createTheme();

// Make generated classNames consistent for snapshots
const generateClassName = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`;

const AllTheProviders = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={msaTheme}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </StylesProvider>
  );
};

const customRender = (component, options, initialRoute = "/") => {
  const componentWithRouter = (
    <Router basename={initialRoute}>{component}</Router>
  );

  return render(componentWithRouter, { wrapper: AllTheProviders, ...options });
};

const renderWithProvider = ({
  childComponent,
  mockStore,
  mockState = {},
  initialRoute,
}) => {
  const store = mockStore || configureMockStore(middlewares)(mockState);

  return {
    ...customRender(
      <Provider store={store}>{childComponent}</Provider>,
      null,
      initialRoute,
    ),
    store,
  };
};

const customRenderHook = (callback, options = {}) => {
  const wrapper = options.wrapper || AllTheProviders;

  return renderHook(callback, { ...options, wrapper });
};

// re-export everything
export * from "@testing-library/react";

// override render and renderHook methods
export {
  customRender as render,
  renderWithProvider,
  customRenderHook as renderHook,
  act as actOnHook,
  userEvent,
};
