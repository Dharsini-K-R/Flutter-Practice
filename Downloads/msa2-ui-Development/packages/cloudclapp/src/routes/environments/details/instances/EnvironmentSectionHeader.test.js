import React from "react";
import { render } from "cloudclapp/src/utils/test-utils";
import EnvironmentSectionHeader from "./EnvironmentSectionHeader";
const props = {
  title: "Development",
  owner: 125,
  status: "OK",
}
jest.mock("cloudclapp/src/components/user-button", () => () => (
  <div>Mock UserButton</div>
));
describe("EnvironmentSectionHeader component", () => {
  it("renders correctly", () => {
    const { container, getByText } = render(
      <EnvironmentSectionHeader {...props} />,
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText("Development")).toBeTruthy();
  });
  it("Calls child component - UserButton", () => {
    const { getByText } = render(
      <EnvironmentSectionHeader {...props} />,
    );
    const mockComponent = getByText("Mock UserButton");
    expect(mockComponent).toBeDefined();
  });
});
