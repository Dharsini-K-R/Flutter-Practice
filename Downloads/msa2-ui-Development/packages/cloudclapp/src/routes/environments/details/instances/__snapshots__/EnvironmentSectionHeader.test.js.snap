// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnvironmentSectionHeader component renders correctly 1`] = `
<div
  class="MuiGrid-root makeStyles-environmentTitleWrapper MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between MuiGrid-grid-xs-12"
>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <div
      class="MuiBox-root MuiBox-root"
    >
      <div
        class="MuiBox-root MuiBox-root"
      >
        <p
          class="MuiTypography-root makeStyles-envName makeStyles-envName MuiTypography-body1"
          id="INSTANCE_DETAILS_ENVIRONMENT_NAME"
        >
          Development
        </p>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <div
      class="MuiBox-root MuiBox-root"
    >
      <div
        class="MuiBox-root MuiBox-root"
      >
        <div>
          Mock UserButton
        </div>
      </div>
    </div>
  </div>
</div>
`;
